components/Autocomplete/index.jsx

```diff
@@ -0,0 +1,260 @@
+import React, {PropTypes} from 'react';
+import Autosuggest from 'react-autosuggest';
+import Scroll from 'react-scroll';
+
+import styles from './index.scss';
+
+const isMobile = window.outerWidth < 550;
+
+export default class Autocomplete extends React.Component {
+  static propTypes = {
+    data: PropTypes.object,
+    disabled: PropTypes.bool,
+    error: PropTypes.string,
+    id: PropTypes.string.isRequired,
+    inputStyle: PropTypes.string,
+    label: PropTypes.string.isRequired,
+    labelHidden: PropTypes.bool,
+    name: PropTypes.string.isRequired,
+    noSuggestionText: PropTypes.string.isRequired,
+    onGetSuggestions: PropTypes.func,
+    placeholder: PropTypes.string,
+    query: PropTypes.object,
+    required: PropTypes.bool,
+    requiredLabel: PropTypes.string,
+    scrollTo: PropTypes.bool,
+    submitOnEnter: PropTypes.bool,
+    suggestionsFooter: PropTypes.any,
+    value: PropTypes.string
+  };
+
+  static defaultProps = {
+    error: null,
+    inputStyle: 'inline',
+    noSuggestionText: 'No results found',
+    placeholder: '',
+    query: {},
+    scrollTo: true
+  }
+
+  state = {
+    noSuggestions: false,
+    showError: false,
+    suggestions: [],
+    value: this.props.value
+  };
+
+  componentWillMount () {
+    this.updateValue(this.props.query[this.props.name] || this.props.value || '');
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.checkIfNoSuggestions(nextProps);
+    if (!this.needsUpdate(nextProps)) return;
+    if (nextProps.error) this.showError();
+    this.updateValue(this.props.query[this.props.name] || nextProps.value || '');
+  }
+
+  onChange = (event, {newValue}) => {
+    this.setState({
+      noSuggestions: false,
+      value: newValue
+    });
+    this.hideError();
+  }
+
+  onFocus = () => {
+    this.scrollToElement();
+  }
+
+  onBlur = () => {
+    this.hideNoSuggestions();
+  }
+
+  onSuggestionsFetchRequested = ({value}) => {
+    this.getSuggestions(value);
+
+    this.setState({
+      suggestions: this.props.data
+    });
+  }
+
+  onSuggestionsClearRequested = () => {
+    this.setState({
+      suggestions: []
+    });
+  };
+
+  getSuggestions = value => {
+    const inputValue = value.trim().toLowerCase();
+    return inputValue.length ?
+      this.props.onGetSuggestions(inputValue) :
+      [];
+  }
+
+  getSuggestionValue = suggestion => suggestion.item
+
+  checkIfNoSuggestions (nextProps) {
+    const noSuggestions = !nextProps.data.items.length && this.state.value.length;
+
+    this.setState({noSuggestions});
+  }
+
+  hideNoSuggestions = () => {
+    this.setState({
+      noSuggestions: false
+    });
+  }
+
+  showError () {
+    this.setState({
+      showError: true
+    });
+  }
+
+  hideError () {
+    this.setState({
+      showError: false
+    });
+  }
+
+  scrollToElement = () => {
+    if (this.props.scrollTo && isMobile) {
+      const scroll = Scroll.animateScroll;
+      scroll.scrollTo(Number(this.node.offsetParent.offsetTop) + 65);
```

Comment:
I would spend some more time trying to figure out where this 65 comes from and whether we can write this in a cleaner way.